hack/shoot-comparator/pkg/shoot/matcher.go

```diff
@@ -55,15 +56,7 @@ func (m *Matcher) Match(actual interface{}) (success bool, err error) {
 		return false, err
 	}
 
-	// Note: we define separate matchers for each field to make input more readable
-	// Annotations are not matched as they are not relevant for the comparison ; both KIM, and Provisioner have different set of annotations
-	for _, matcher := range []propertyMatcher{
-		// We need to skip comparing type meta as Provisioner doesn't set it.
-		// It is simpler to skip it than to make fix in the Provisioner.
-		//{
-		//	GomegaMatcher: gomega.BeComparableTo(eShoot.TypeMeta),
-		//	expected:      aShoot.TypeMeta,
-		//},
+	matchers := []propertyMatcher{
 		{
 			GomegaMatcher: gomega.BeComparableTo(eShoot.Name),
 			expected:      aShoot.Name,
```

Comment:
We initialise `expected` field with `actual` value. Should the field `propertyMatcher.expected` be called `propertyMatcher.actual`

Reply:
I think you are right, the name is misleading. What we actually do is:
- compare if gardener shoot generated by the converter is contained in shoot stored on cluster.

The `actual` should point to shoot stored on cluster and the `expected` to shoot generated by the converter. We can also think of changing the names e.g. `stored` and `converted`.